src/linker/Linker.Analysis/README.md

```diff
@@ -0,0 +1,61 @@
+# Linker analyzer
+
+We have been prototyping a callgraph-based analyzer for the linker, that reports "linker-unsafe" usage in the IL reachable from some set of entry points (Main, or all public surface area). It simulates attributes that influence how unsafe patterns get reported.
+
+# Instructions
+
+To give it a try:
+
+## Clone and build
+```
+> git clone https://github.com/vitek-karas/linker -b Analysis
+> cd linker
+linker> ./eng/dotnet.{sh/ps1} build illink.sln
+```
+
+## Run
+
+Run the linker specifying the desired set of roots, optimizations, etc. A summary of the analysis will be written to stdout, and the "callstacks" will be written to `trimanalysis.json` in the linker output directory.
+For example: 
+
+```
+linker> dotnet ./src/ILLink.Tasks/bin/Debug/netcoreapp2.0/illink.dll -a console -c link -u link -d testapps/console/out/ --skip-unresolved true --ignore-descriptors true --verbose --dump-dependencies --used-attrs-only true -out testapps/console/linked --analyze-for-trim
+```
+
+Here, the command-line flags have the following meaning.
+```
+-a console:                    Roots console.dll. The linker will see that it is an application, and root only the entry point (not all public APIs)
+-c link -u link:               Tells the linker to trim all application (-u) and framework (-c) libraries unless otherwise specified.
+-d testapps/console/out:       Tells the linker where to find the dependencies of the application.
+--skip-unresolved:             Continue when encountering unresolved dependencies.
+--ignore-descriptors true:     Don't root types/members specified in embedded .xml descriptors. Use this option to see only what is kept by the linker's analysis.
+--verbose:                     Output extra info to stdout, such as the methods with unanalyzed reflection patterns.
```

Comment:
This currently doesn't work with --analyze-for-trim - I'll see about fixing it.

Reply:
On second thought - I'm not sure we should write these out with --analyze-for-trim. Mainly because of the method is marked as safe (through some other means), then the fact that linker doesn't understand it is sort of "by design". So we should not report it.
And if it is not marked by anything, we will report the method as unsafe regardless. Although in this case I could see some usefulness in reporting why the linker was not able to figure it out.

Reply:
Ah, ok. I don't have any particular preference on whether we should continue writing them out, I just meant to describe the existing behavior, but this was inaccurate. Currently --verbose seems to just output (when --analyze-for-trim is true) info about the output actions, and the understood patterns are always output (with --analyze-for-trim). Maybe we should turn off the understood pattern output too?

Reply:
I added those just out of curiousity - that should definitely go.
It should not output unrecognized patterns with analysis on (the code simply doesn't exist there).